addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/handler/AccountHandler.java

```diff
@@ -0,0 +1,635 @@
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.openhab.binding.amazonechocontrol.internal.AccountConfiguration;
+import org.openhab.binding.amazonechocontrol.internal.Connection;
+import org.openhab.binding.amazonechocontrol.internal.ConnectionException;
+import org.openhab.binding.amazonechocontrol.internal.LoginServlet;
+import org.openhab.binding.amazonechocontrol.internal.StateStorage;
+import org.openhab.binding.amazonechocontrol.internal.discovery.AmazonEchoDiscovery;
+import org.openhab.binding.amazonechocontrol.internal.discovery.IAmazonEchoDiscovery;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonBluetoothStates;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonBluetoothStates.BluetoothState;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonDevices.Device;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonFeed;
+import org.openhab.binding.amazonechocontrol.internal.jsons.JsonSmartHomeDevice;
+import org.osgi.service.http.HttpService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Handles the connection to the amazon server.
+ *
+ * @author Michael Geramb - Initial Contribution
+ */
+public class AccountHandler extends BaseBridgeHandler implements IAmazonEchoDiscovery {
+
+    private final Logger logger = LoggerFactory.getLogger(AccountHandler.class);
+    private StateStorage stateStorage;
+    private AccountConfiguration config;
+    private Connection connection;
+    private List<EchoHandler> echoHandlers = new ArrayList<>();
+    private List<SmartHomeBaseHandler> smartHomeHandlers = new ArrayList<>();
+    private List<FlashBriefingProfileHandler> flashBriefingProfileHandlers = new ArrayList<>();
+    private Object synchronizeConnection = new Object();
+    private Map<String, Device> jsonSerialNumberDeviceMapping = new HashMap<>();
+    private ScheduledFuture<?> refreshJob;
+    private ScheduledFuture<?> refreshLogin;
+    private boolean updateSmartHomeDeviceList;
+    private boolean discoverFlashProfiles;
+    private boolean smartHodeDeviceListEnabled;
+    private String currentFlashBriefingJson = "";
+    private HttpService httpService;
+    private LoginServlet loginServlet;
+
+    public AccountHandler(@NonNull Bridge bridge, @NonNull HttpService httpService) {
```

Comment:
Please do not use NonNull but use NonNullByDefault as [described in the coding guidelines](https://www.eclipse.org/smarthome/documentation/development/guidelines.html#a-code-style)

Reply:
Ok